corgi/api/views.py

```diff
@@ -400,6 +401,47 @@ def list(self, request, *args, **kwargs):
         response["Location"] = f"/api/{CORGI_API_VERSION}/components/{component.uuid}"
         return response
 
+    @action(methods=["put"], detail=True)
+    def olcs_test(self, request, uuid=None):
+        """Allow OpenLCS to upload copyright text / license scan results for a component"""
+        # In the future these could be separate endpoints
+        # For testing we'll just keep it under one endpoint
+        if utils.running_prod():
+            # This is only temporary for OpenLCS testing
+            # Do not enable in production until we add OIDC authentication
+            return Response(status=403)
+        component = self.queryset.filter(uuid=uuid).first()
+        if not component:
+            return Response(status=404)
+
+        copyright_text = request.data.get("copyright_text")
+        license_concluded = request.data.get("license_concluded")
+        openlcs_scan_url = request.data.get("openlcs_scan_url")
+        openlcs_scan_version = request.data.get("openlcs_scan_version")
+        if (
+            not copyright_text
+            and not license_concluded
+            and not openlcs_scan_url
+            and not openlcs_scan_version
+        ):
+            # At least one of above is required, else Bad Request
+            return Response(status=400)
```

Comment:
I think this should be a 200 actually. It's essentially a PUT that changes nothing, but that is not an invalid operation, simply a noop one. It's analogous to a PUT that submits the text that is already in a particular field.

Reply:
I'll probably keep this unchanged for now given time constraints, and since I already told the OpenLCS team at least one of these fields is required. I also think it's good for a beta-test, to explicitly help the user realize "none of the fields you submitted were recognized", e.g. if there was a typo in the JSON data / field name they submitted.

We can definitely change it for the final implementation though.